pandas/tseries/holiday.py

```diff
@@ -159,24 +166,34 @@ def __init__(
         year=None,
         month=None,
         day=None,
-        offset=None,
-        observance=None,
+        offset: BaseOffset | list[BaseOffset | list[BaseOffset]] | None = None,
```

Comment:
I don't think `list[list[BaseOffset]]` should be supported. It's not worth the complexity when the provided `list[BaseOffset]` can be spat into a new argument

Reply:
I removed the logic for handling this case and instead only throw a more informative error now. The type hint alone might already be sufficient for when someone chains `Holiday.offset` without considering that they can be lists, but I think the more detailed exception could also be nice to have.